components/frontend/src/hooks/url_search_query.js

```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react';
+import history from 'history/browser';
+
+const registeredURLSearchQueryKeys = new Set(["report_date", "report_url", "hide_toasts"]);
+
+export function registeredURLSearchParams() {
+    // Return registered URL search parameters only; to prevent CodeQL js/client-side-unvalidated-url-redirection
```

Comment:
```suggestion
    // Return registered URL search parameters only; to prevent untrusted URL redirection by client
```

Reply:
I left the CodeQL message in there in case anyone is searching the code base with that message id. But I'll add the regular English text as well.